pydocx/models.py

```diff
@@ -120,10 +121,25 @@ class ParkingLot(XmlModel):
     def __init__(self, *types, **kwargs):
         default = kwargs.pop('default', [])
         super(XmlCollection, self).__init__(self, default=default)
-        self.types = set(types)
+        self._types = types
         self._name_to_type_map = None
 
     @property
+    def types(self):
+        return set(self._set_types(*self._types))
+
+    def _set_types(self, *types):
```

Comment:
My research leads me to believe that no caching is needed (it won't actually change anything).